README.md

```diff
@@ -170,6 +170,28 @@ class PostSerializer < ActiveModel::Serializer
 end
 ```
 
+### Context-aware predicate methods
+
+If you want to serialize attributes or associations differently depending on the context they're serialized, you can use:
+
+```ruby
+class PostSerializer < ActiveModel::Serializer
+  attributes :id, :body
+
+  has_many :comments
+
+  def body
+    if listed? or associated? # same as `if nested?`
+      object.body.lines.first + '...'
+    else
+      object.body
+    end
+  end
+end
+```
```

Comment:
Originally, yes. Most of it ended up being lots of boilerplate, creating many classes and scattering them throughout my code to achieve a fairly common behaviour: essentially, unless you're serializing a single object for display, only serialize enough information to get to that canonical full-details view, with a few extra display parameters to generate links.

On the other hand, as I've updated to `v0.10.0`, adapters have really helped handle that for me. They provide a better abstraction to handle these concerns; and I no longer think this patch is useful. 

Updating my patch helped keep my tests green, but in the last few days I've discarded it after years of it serving me well. All of the motivations for it, #801, and #498 are covered with adapters, and I, for one, welcome our new `v0.10.0` overlords. 

:keycap_ten:
:raised_hands:
:smile: